deploy/docker/notebook/base/base.yaml

```diff
@@ -472,11 +479,16 @@ addfiles:
                   os_old_path = str(Path().resolve()) + '/' + os_old_path
                   os_new_path = str(Path().resolve()) + '/' + os_new_path
 
-                  if not (self.is_inside_path(os_new_path, self.work_path) or self.is_inside_path(os_new_path, self.shared_path)):
-                      return
+                  #If the new path doesn't include the shared or work path, delete the old path. We only track work_path and shared_path
+                  if not (self.is_inside_path(os_new_path) or self.is_inside_path(os_new_path)):
```

Comment:
I think, your logic is missing some cases. Let's consider all possible scenarios for the `old_path` and the `new_path` as it relates to them being inside (✅) or outside (❌) the tracked paths. Below is the table of the outcomes and the actions the code should make:
<table>
<tbody>
<tr>
<td colspan="2" rowspan="2"></td>
<td colspan="2">old_path</td>
</tr>
<tr>
<td>❌</td>
<td>✅</td>
</tr>
<tr>
<td rowspan="2">new_path</td>
<td>❌</td>
<td>return</td>
<td>Delete</td>
</tr>
<tr>
<td>✅</td>
<td>Save</td>
<td>Rename</td>
</tr>
</tbody>
</table>

Reply:
Because there is a check for existence before the create or delete methods, the logic is more like this, where the rename function simply calls the delete function then the create/update file function

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:////Users/swazoo.claybonlabshare.org/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip.htm">
<link rel=File-List
href="file:////Users/swazoo.claybonlabshare.org/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip_filelist.xml">
<style>
<!--table
	{mso-displayed-decimal-separator:"\.";
	mso-displayed-thousand-separator:"\,";}
@page
	{margin:.75in .7in .75in .7in;
	mso-header-margin:.3in;
	mso-footer-margin:.3in;}
tr
	{mso-height-source:auto;}
col
	{mso-width-source:auto;}
br
	{mso-data-placement:same-cell;}
td
	{padding-top:1px;
	padding-right:1px;
	padding-left:1px;
	mso-ignore:padding;
	color:black;
	font-size:12.0pt;
	font-weight:400;
	font-style:normal;
	text-decoration:none;
	font-family:Calibri, sans-serif;
	mso-font-charset:0;
	mso-number-format:General;
	text-align:general;
	vertical-align:bottom;
	border:none;
	mso-background-source:auto;
	mso-pattern:auto;
	mso-protection:locked visible;
	white-space:nowrap;
	mso-rotate:0;}
.xl65
	{color:windowtext;
	font-size:14.0pt;
	font-family:Helvetica, sans-serif;
	mso-font-charset:0;}
-->
</style>
</head>

<body link="#0563C1" vlink="#954F72">



 file path location | save | delete
-- | -- | --
no change | x | x
move inside tracked paths | x |  
move outside tracked paths |   | x
move within tracked paths | x | x



</body>

</html>

which for the rename hook means:
If the new path is not in the tracked paths, delete the old path.
If the new path is inside the tracked paths, create the new path and disregard the old path since it does not exist in our db.